src/Api/PubnubApi/EndPoint/PubSub/PublishOperation.cs

```diff
@@ -205,67 +205,80 @@ internal void Publish(string channel, object message, bool storeInHistory, int t
-                    if (publishStatus == 1)
+                    List<object> result = ProcessJsonResponse(requestState, json);
+
+                    if (result != null && result.Count >= 3)
                     {
-                        ProcessResponseCallbacks(result, requestState);
+                        int publishStatus;
+                        Int32.TryParse(result[0].ToString(), out publishStatus);
+                        if (publishStatus == 1)
+                        {
+                            ProcessResponseCallbacks(result, requestState);
+                        }
+                        else
+                        {
+                            PNStatusCategory category = PNStatusCategoryHelper.GetPNStatusCategory(400, result[1].ToString());
+                            PNStatus status = new StatusBuilder(config, jsonLibrary).CreateStatusResponse<PNPublishResult>(PNOperationType.PNPublishOperation, category, requestState, 400, new PNException(json));
+                            if (requestState.PubnubCallback != null)
+                            {
+                                requestState.PubnubCallback.OnResponse(default(PNPublishResult), status);
+                            }
+                        }
                     }
                     else
                     {
-                        PNStatusCategory category = PNStatusCategoryHelper.GetPNStatusCategory(400, result[1].ToString());
-                        PNStatus status = new StatusBuilder(config, jsonLibrary).CreateStatusResponse<PNPublishResult>(PNOperationType.PNPublishOperation, category, requestState, 400, new PNException(json));
-                        if (requestState.PubnubCallback != null)
-                        {
-                            requestState.PubnubCallback.OnResponse(default(PNPublishResult), status);
-                        }
+                        ProcessResponseCallbacks(result, requestState);
                     }
                 }
-                else
+
+                CleanUp();
+            }
+            catch (Exception ex)
+            {
+                PNStatusCategory category = PNStatusCategoryHelper.GetPNStatusCategory(400, ex.ToString());
+                PNStatus status = new StatusBuilder(config, jsonLibrary).CreateStatusResponse(PNOperationType.PNPublishOperation, category, requestState, 400, new PNException(ex.ToString()));
+                if (requestState.PubnubCallback != null)
                 {
-                    ProcessResponseCallbacks(result, requestState);
+                    requestState.PubnubCallback.OnResponse(default(PNPublishResult), status);
                 }
-            }
 
-            CleanUp();
+            }
         }
 
         internal async Task<PNResult<PNPublishResult>> Publish(string channel, object message, bool storeInHistory, int ttl, Dictionary<string, object> metaData, Dictionary<string, object> externalQueryParam)
```

Comment:
Most of the code is repeated in this method

Reply:
yeah..due to async/await and callback style outputs, some code was repeated due to threading issues. In the current PR  only catch block section code was added to handle unknown exception which customer could not share. Remaining code nothing changed.